### PR TITLE
Adding tests for the app.rb file.

### PR DESCRIPTION
**Pull request by DavidEBest**
- Added tests to cover the app.rb file
- Made the ComplianceViewer class more testable by allowing ComplianceData to be passed in.
- Corrected a few syntax issues exposed by the new .rubocop.yml.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -10,7 +10,16 @@ class ComplianceViewer < Sinatra::Base
   register Sinatra::ConfigFile
   include ZapReport
 
-  config_file 'credentials.yml'
+  if settings.environment == :test
```
Comment (ctro) [inline app.rb]: would this be better if we required `credentials/test.yml` and `credentials/development.yml`, etc?  That way we could just load yml for the environment.

File: spec/stub_classes.rb [hunk 2]
```diff
@@ -16,6 +16,23 @@ def initialize(alerts = [])
     end
   end
 
+  class StubObjectSummary
```
Comment (ctro) [inline spec/stub_classes.rb:19]: This stub class is getting more complicated... I've had good luck with Mocha in the past:
https://github.com/freerange/mocha#quick-start

**Comment by ctro** [comment]
Yay tests!
Looks good, left a couple small comments.

File: app.rb (hunk 1 above)
Comment (DavidEBest) [inline app.rb]: I like that idea. I'll give it a shot.

File: spec/stub_classes.rb (hunk 2 above)
Comment (DavidEBest) [inline spec/stub_classes.rb:19]: I'll give it a shot.

File: .rubocop.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+  # inline conditionals don't get picked up by simplecov, so preferable to avoid them
+  Enabled: false
+Style/IndentArray:
+  Enabled: false
+Style/RedundantBegin:
+  Enabled: false
+Style/SignalException:
+  Enabled: false
+Style/StringLiterals:
+  Enabled: false
```
Comment (afeld) [inline .rubocop.yml]: /me is now tempted to make an `afeld_rubocop` gem with [shared configuration](https://github.com/bbatsov/rubocop#inheriting-configuration-from-a-dependency-gem) :wink: 

Actually, maybe we should [inherit from the other repository](https://github.com/bbatsov/rubocop#inheriting-configuration-from-a-remote-url) so we don't have to maintain the list in two places:

``` yaml
inherit_from:
- https://raw.githubusercontent.com/18F/concourse-compliance-testing/rubocop-config/.rubocop.yml
```

replacing `rubocop-config` with `master` once https://github.com/18F/concourse-compliance-testing/pull/21 is merged.

File: app.rb [hunk 4]
```diff
@@ -43,7 +52,7 @@ def initialize
 
   get '/results/:name' do |name|
     versions = compliance_data.versions name
-    'Invalid Project' if versions.count == 0
+    return 'Invalid Project' if versions.count == 0
```
Comment (afeld) [inline app.rb]: Minor: I'd use `if versions.empty?`.

http://ruby-doc.org/core-2.2.0/Array.html#method-i-empty-3F

**Comment by DavidEBest** [comment]
This one should be good for a re-review. @ctro - good call on `mocha`; it made it much easier to test `ComplianceData` without relying on an overly complex stub.

File: spec/app_spec.rb [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+require 'stub_classes'
+
+describe 'ComplianceViewer' do
+  include Rack::Test::Methods
+
+  class StubComplianceData
+  end
+
+  def app
+    ComplianceViewer.new! StubComplianceData.new
```
Comment (afeld) [inline spec/app_spec.rb:13]: What does `new!` do?

File: spec/app_spec.rb [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+      get '/results/name'
+      expect(last_response.redirect?).to eq true
+      expect(last_response['Location']).to include('/auth/myusa')
+    end
+
+    it 'redirects to the auth page if unauthed without a name' do
+      get '/results/'
+      expect(last_response.redirect?).to eq true
+      expect(last_response['Location']).to include('/auth/myusa')
+    end
```
Comment (afeld) [inline spec/app_spec.rb:82]: Nitpick: this is a different route, so I'd put it in a `describe '/results'` block.

**Comment by afeld** [comment]
Just a minor thing—feel free to merge yourself after that!

File: spec/app_spec.rb (hunk 6 above)
Comment (DavidEBest) [inline spec/app_spec.rb:82]: This is intentional. The way routing works for Sinatra apps, `/results` and `/results/` are different, with `/results/` going to `/results/{name}`. That's why I put it in this block.

Comment (afeld) [inline spec/app_spec.rb:82]: No kidding!? Good to know.

File: spec/app_spec.rb (hunk 5 above)
Comment (DavidEBest) [inline spec/app_spec.rb:13]: Took me a while to figure that one out. :) Sinatra redefines `new` to include the Rack middleware. Before doing that, it aliases the existing functionality to `new!`. I do that so I can get access an actual `ComplianceViewer` instance.

https://github.com/sinatra/sinatra/blob/v1.3.3/lib/sinatra/base.rb#L1369-1370

Comment (afeld) [inline spec/app_spec.rb:13]: Hoo boy :sweat_smile: Mind adding that as an inline comment? If the tests weren't passing, I would have assumed it was a typo.

Comment (ctro) [inline spec/app_spec.rb:13]: Wow, eeeeeenteresting.  I did not know that either.

Comment (DavidEBest) [inline spec/app_spec.rb:13]: Yeah. Seemed pretty squirrelly. I added a comment here: https://github.com/18F/compliance-viewer/pull/10

**Pull request merged**